File: src/main/java/com/ada/conversation/context/MemoriesContextItem.java
package com.ada.conversation.context;

import com.ada.conversation.application.dto.*;
import com.ada.shared.observability.MeasuredContextItem;
import org.springframework.core.annotation.Order;
import org.springframework.stereotype.Component;

@Component
@Order(40)
@MeasuredContextItem("memories")
public class MemoriesContextItem implements ContextItem {
  public LlmContentComponent component() {
    return LlmContentComponent.MEMORIES;
  }

  public ContextState apply(ChatRequest r, ContextState c) {
    return c;
  }
}

File: src/main/kotlin/com/ada/conversation/context/MemoriesContextItem.kt
package com.ada.conversation.context

import com.ada.conversation.application.dto.ChatRequest
import com.ada.conversation.application.dto.ContextState
import com.ada.conversation.application.dto.LlmContentComponent
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Component
import com.ada.shared.observability.MeasuredContextItem

@Component
@Order(40)
@MeasuredContextItem("memories")
class MemoriesContextItem : ContextItem {
    override val component = LlmContentComponent.MEMORIES

    override fun apply(request: ChatRequest, current: ContextState): ContextState = current
}
